clang/include/clang/Sema/Sema.h

```diff
@@ -13163,6 +13163,18 @@ class Sema final {
                                    SourceLocation BuiltinLoc,
                                    SourceLocation RParenLoc);
 
+  template <typename AttrTy> bool isDecoratedWithSyclAttribute(QualType Ty) {
```

Comment:
Could you please (if you have some spare time) also refactor SemaSYCL to use this function instead of `isSyclDeviceGlobalType` added by me there?